CONTRIBUTION.md

```diff
@@ -71,6 +71,8 @@ git tools.
   - Format your code and fix all the spelling/grammatical mistakes.
   - Limit the column length of the code and the comments within the code to 120 characters.
   - Error messages within the code should be limited to a single line.
+  - Update the documentation if your PR is introducing or changing user facing functionality.
+  - Ensure each PR commit compiles and all unit and e2e tests pass on your local machine.
```

Comment:
Not sure, if we want to go this route

Reply:
Which route? e2e or compile? I definitely want each commit to compile.

Reply:
e2e tests pass - say if we've a vendor commit and the changes it brings in other commit

Reply:
I am fine having an exception when we are introducing vendor commits. So we can add a line mentioning that. But in general I don't want to see commits that don't compile.

Reply:
Do you want to remove `e2e test` or having another bullet point saying that ` pass e2e test is optional for vendor commit but good to have`?

Reply:
There should be a sub bullet stating that PRs that have vendor commits are an exception to the rule.

Reply:
Sub bullet added.